admin/services/data-access/sql.service.js

```diff
@@ -134,172 +148,156 @@ async function generateParams (tableName, data) {
     if (cacheData.dataType === 'Decimal' || cacheData.dataType === 'Numeric') {
       options.precision = cacheData.precision
       options.scale = cacheData.scale
+    } else if (cacheData.maxLength) {
+      options.length = cacheData.maxLength
     }
+
     logger.debug(`sql.service: generateParams: options set for [${column}]`, options)
     params.push({
       name: column,
       value,
-      type: R.prop(findTediousDataType(cacheData.dataType), TYPES),
+      type: R.prop(findDataType(cacheData.dataType), sqlService.TYPES),
       options
     })
   }
   return params
 }
 
-function parseResults (results) {
-  // omit metadata for now, can introduce if useful at later date
-  const jsonArray = []
-  results.forEach(row => {
-    const json = {}
-    // const metadata = []
-    row.forEach(col => {
-      if (col.metadata.colName !== 'version') {
-        if (col.metadata.type.type === 'DATETIMEOFFSETN' && col.value) {
-          json[col.metadata.colName] = moment(col.value)
-        } else {
-          json[col.metadata.colName] = col.value
-        }
-      }
-    })
-    jsonArray.push(json)
-  })
-  return jsonArray
+/** SQL Service **/
+const sqlService = {
+  // SQL type-mapping adapter.  Add new types as required.
+  TYPES: {
```

Comment:
@jon-shipley really optional but wouldn't you prefer the TYPES being stored separately and imported into the sql.service?

Reply:
@GeorgeChatzigiannis - why?

Reply:
@jon-shipley it would seem cleaner to me and easier to maintain but again it’s a matter of preference.